### PR TITLE
guide.md: update Q&A in "Dictionaries Challenges" section

### PR DESCRIPTION
**Pull request by NPalopoli**
Updated question and answer in section **Dictionaries Challenges** based on corresponding section in episode [01-short-introduction-to-Python.md](https://github.com/datacarpentry/python-ecology-lesson/blob/gh-pages/_episodes/01-short-introduction-to-Python.md)

### CODE REVIEW

File: _extras/guide.md [hunk 1]
```diff
@@ -40,12 +40,19 @@ installation. Otherwise, it will tell you that the system is good to go and read
 	`tuple`
 
 ### Dictionaries Challenges
-* Can you do reassignment in a dictionary? Give it a try.
+
+* Changing dictionaries: 2. Reassign the second value.
```
Comment (maxim-belkin) [inline _extras/guide.md]: 1. What does `2.` mean?
2. `Second value` does not sound right as dictionary is not a sequence of values but of key-value pairs.

Comment (wrightaprilm) [reply _extras/guide.md]: I agree with this "Reassign the value of the second key" might be what we're looking for here.

Comment (maxim-belkin) [reply _extras/guide.md]: Thank you, April. However, we can not even use "the second key" as dictionaries are not ordered. We have to be very specific with dictionaries and say `value that corresponds to key of <blah>` (blah = 2 in this case).

Comment (wrightaprilm) [reply _extras/guide.md]: Good catch - easy to forget. I like

"Reassign the value that corresponds to the key "2"."

**Comment by NPalopoli** [comment]
Nice one! Thanks Maxim and April. I've created #336 to address this in the
lesson.

--
Nicolás Palopoli
Structural Bioinformatics Group
Universidad Nacional de Quilmes - CONICET
http://orcid.org/0000-0001-7925-6436

File: _extras/guide.md (hunk 1 above)
Comment (maxim-belkin) [reply _extras/guide.md]: Please don't hate my but... no double quotes around 2 because `2` and `"2"` are two different and legitimate keys.

**Comment by NPalopoli** [comment]
I agree, thanks! And I think we can avoid the double quotes without
allowing for confusions.

I'm not sure if the previous changes have been accepted or not yet. Please
go ahead with the PR for removing the double quotes and close the issue if
done.

File: _extras/guide.md (hunk 1 above)
Comment (wrightaprilm) [reply _extras/guide.md]: Resolved by #336

Comment (wrightaprilm) [reply _extras/guide.md]: My mistake, this aspect is not resolved by #336. Could we change this  to 

" Reassign the value that corresponds to the key `2`."

Comment (NPalopoli) [reply _extras/guide.md]: Thank you @wrightaprilm and @maxim-belkin. Should be correct now.

**Comment by wrightaprilm** [comment]
Great, thanks for fixing that. I'm happy if you're happy, @maxim-belkin.

File: _extras/guide.md [hunk 2]
```diff
@@ -40,12 +40,19 @@ installation. Otherwise, it will tell you that the system is good to go and read
 	`tuple`
 
 ### Dictionaries Challenges
-* Can you do reassignment in a dictionary? Give it a try.
 
-Make sure it is also clear that access to 'the second value' is actually just about the key name. Add for example `rev[10] = "ten"` to clarify it is not about the position.
+* Changing dictionaries: 2. Reassign the value that corresponds to the key `2`.
```
[the remark this answers is not in the document]
Comment (wrightaprilm) [reply _extras/guide.md]: There's a one in the lesson, but no one has apparently felt moved to add it to the guide (it's just printing the dictionary).

[the remark this answers is not in the document]
Comment (maxim-belkin) [reply _extras/guide.md]: OK, I see its purpose now. Let's merge this as is then. I'll submit my suggestions (at some point)

**Comment by wrightaprilm** [comment]
Thanks @NPalopoli!

**Pull request merged**